SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -318,9 +318,22 @@ void ASpatialFunctionalTest::AddStepBlueprint(const FString& StepName, const FWo
 	StepDefinitions.Add(StepDefinition);
 }
 
-void ASpatialFunctionalTest::AddGenericStep(const FSpatialFunctionalTestStepDefinition& StepDefinition)
+void ASpatialFunctionalTest::AddStepFromDefinition(const FSpatialFunctionalTestStepDefinition& StepDefinition, const FWorkerDefinition& Worker)
 {
-	StepDefinitions.Add(StepDefinition);
+	FSpatialFunctionalTestStepDefinition StepDefinitionCopy = StepDefinition;
+
+	StepDefinitionCopy.Workers.Add(Worker);
+
+	StepDefinitions.Add(StepDefinitionCopy);
+}
+
+void ASpatialFunctionalTest::AddStepFromDefinitionMulti(const FSpatialFunctionalTestStepDefinition& StepDefinition, TArray<FWorkerDefinition> Workers)
```

Comment:
Is this meant to be passed by value? I guess we would just copy it anyway... then is there a reason for the above one to be passed by reference?

Reply:
My bad, corrected